src/charm.py

```diff
@@ -1,303 +1,309 @@
-
-        gen_certs(model, hookenv.service_name())
-
-        ca_bundle = b64encode(Path("/run/cert.pem").read_bytes()).decode("utf-8")
-
-        self.model.pod.set_spec(
-            {
-                "version": 3,
-                "serviceAccount": {
-                    "roles": [
-                        {
-                            "global": True,
-                            "rules": [
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": ["poddefaults"],
-                                    "verbs": [
-                                        "get",
-                                        "list",
-                                        "watch",
-                                        "update",
-                                        "create",
-                                        "patch",
-                                        "delete",
-                                    ],
-                                },
-                            ],
-                        }
-                    ],
+    @property
+    def container(self):
+        """Return container."""
+        return self._container
+
+    @property
+    def k8s_resource_handler(self):
+        """Update K8S with K8S resources."""
+        if not self._k8s_resource_handler:
+            self._k8s_resource_handler = KubernetesResourceHandler(
+                field_manager=self._lightkube_field_manager,
+                template_files=K8S_RESOURCE_FILES,
+                context=self._context,
+                logger=self.logger,
+            )
+        load_in_cluster_generic_resources(self._k8s_resource_handler.lightkube_client)
+        return self._k8s_resource_handler
+
+    @k8s_resource_handler.setter
+    def k8s_resource_handler(self, handler: KubernetesResourceHandler):
+        self._k8s_resource_handler = handler
+
+    @property
+    def crd_resource_handler(self):
+        """Update K8S with CRD resources."""
+        if not self._crd_resource_handler:
+            self._crd_resource_handler = KubernetesResourceHandler(
+                field_manager=self._lightkube_field_manager,
+                template_files=CRD_RESOURCE_FILES,
+                context=self._context,
+                logger=self.logger,
+            )
+        load_in_cluster_generic_resources(self._crd_resource_handler.lightkube_client)
+        return self._crd_resource_handler
+
+    @property
+    def _admission_webhook_layer(self) -> Layer:
+        """Create and return Pebble framework layer."""
+        layer_config = {
+            "summary": "admission-webhook layer",
+            "description": "Pebble config layer for admission-webhook-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Entry point of admission-webhook-operator image",
```

Comment:
I think it is not an Entry

that is the description of the service, no?

Reply:
Yeah, it is being copied from some earlier sidecar charm and it is spread all over. Let's change it.

Reply:
May be at the top of the charm:
```
METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
```
And then we can use it:
```
            "services": {
                self._container_name: {
                    "override": "replace",
                    "summary": METADATA["summary"],
```

Reply:
I think summary for the charm and for the service is different :)
at least in other charms it is done differently

Reply:
Yes. Agreed. There could be more than one container per charm. Need to come up with summary for this one.